mediorum/server/server.go

```diff
@@ -142,6 +142,9 @@ func New(config MediorumConfig) (*MediorumServer, error) {
 	echoServer.HideBanner = true
 	echoServer.Debug = true
 
+	// CORS should only be in the `routes` group, but we need it here because /mediorum is rewritten to /
+	echoServer.Use(middleware.CORS())
```

Comment:
@stereosteve CORS is probably not a big deal with this use case, right? because this isn't Identity Service and we'd be using JWT-style auth (if any) for uploads/playback anyway. also, there's no one single client that we'd want to use as the allowed origin ("decentralized", etc...)

Reply:
okay so this was to fix CORS not being set when rewritten, but I now see your comment about not doing that because then the Express server also sets the header. the actual problem seems to be that the reverse proxy is catching the browser's OPTIONS preflight request instead of allowing the CORS middleware to catch it (confirmed in the logs that it's going to the upstream container instead of mediorum). I'm adding an OPTIONS handler explicitly for the /uploads route for now